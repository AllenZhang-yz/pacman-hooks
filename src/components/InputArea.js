import React from "react";

const InputArea = ({ onInputSubmit, handleChange, inputCommand }) => {
  return (
    <form className="input-area" onSubmit={e => onInputSubmit(e)}>
      <input
        placeholder="Please enter command"
        onChange={e => handleChange(e.target.value)}
        value={inputCommand}
      />
    </form>
  );
};

export default InputArea;
