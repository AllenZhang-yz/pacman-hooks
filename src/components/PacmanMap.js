import React from "react";
import Pacman from "./Pacman";

class PacmanMap extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      girdList: [],
      isShowPacman: false
    };
  }

  componentDidMount() {
    const { gridNumber } = this.props;
    let newGridList = [];
    newGridList.length = gridNumber * gridNumber;
    newGridList.fill("");
    this.setState({
      girdList: newGridList
    });
  }

  isPacmanPosition = index => {
    const { gridNumber, position} = this.props;
    const pacmanIndex = position.y * gridNumber + position.x;

    return index === pacmanIndex;
  };

  render() {
    const { direction, gridNumber } = this.props;
    const { girdList } = this.state;

    return (
      <div className="pacman-map">
        {girdList.map((grid, index) => (
          <div key={index} className="pacman-map__cell">
            {`(${index % gridNumber}, ${parseInt(index / gridNumber)})`}
            {this.isPacmanPosition(index) ? (
              <Pacman direction={direction} />
            ) : null}
          </div>
        ))}
      </div>
    );
  }
}

export default PacmanMap;
