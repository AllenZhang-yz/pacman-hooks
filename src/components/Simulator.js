import React from "react";
import InputArea from "./InputArea";
import PacmanMap from "./PacmanMap";

const directions = ["NORTH", "EAST", "SOUTH", "WEST"];
const directionMove = {
  NORTH: { moveX: 0, moveY: 1 },
  EAST: { moveX: 1, moveY: 0 },
  SOUTH: { moveX: 0, moveY: -1 },
  WEST: { moveX: -1, moveY: 0 }
};
const initialState = {
  isReport: false,
  gridNumber: 5,
  direction: "",
  position: {
    x: -1,
    y: -1
  },
  inputCommand: "",
  message: ""
};

class Simulator extends React.Component {
  constructor() {
    super();
    this.state = initialState;
  }

  // set input value when onChange
  handleChange = value => {
    this.setState({
      inputCommand: value
    });
  };

  // reset initialState when click RESET Button
  handleClick = () => {
    this.setState(initialState);
  };

  // press 'enter' on keyboard
  onInputSubmit = e => {
    e.preventDefault();
    const { inputCommand, position, isReport } = this.state;
    // reset message state and
    this.setState({
      message: ""
    });

    if (isReport) {
      this.setState({
        message: "Please click RESET button to start a new test!"
      });
      return;
    }

    // eslint-disable-next-line
    let commands = inputCommand.toUpperCase().split(/\s|\,/);

    if (!this.availablePosition(position) && commands[0] !== "PLACE") {
      // ErrorMsg: invalid command
      this.setState({
        message: "Please enter place command FIRST!"
      });
      return;
    }
    if (commands[0] === "PLACE" && commands.length === 4) {
      this.placePacman(commands.slice(1));
      return;
    }
    if (commands[0] === "MOVE") {
      this.movePacman();
      return;
    }
    if (commands[0] === "LEFT" || commands[0] === "RIGHT") {
      this.rotatePacman(commands[0]);
      return;
    }
    if (commands[0] === "REPORT") {
      this.report();
      return;
    }

    // ErrorMsg: invalid command
    this.setState({
      message: "Invalid Command!"
    });
  };

  getIndexFromArray = (value, arr) => {
    return arr.indexOf(value);
  };

  // check is position in map
  availablePosition = position => {
    const { gridNumber } = this.state;
    if (position.x < 0 || position.x >= gridNumber) {
      return false;
    }
    if (position.y < 0 || position.y >= gridNumber) {
      return false;
    }

    return true;
  };

  // place command
  placePacman = ([x, y, face]) => {
    const { gridNumber } = this.state;
    x = x - 0;
    y = y - 0;
    if (!this.availablePosition({ x, y })) {
      // ErrorMsg: invalid place
      this.setState({
        message: `Invalid Position, Please place position X and Y in 0 ~ ${gridNumber -
          1}`
      });
      return;
    }

    if (this.getIndexFromArray(face, directions) === -1) {
      // ErrorMsg: invalid direction
      this.setState({
        message: `Invalid Direction, Please use NORTH, EAST, SOUTH, WEST`
      });
      return;
    }

    this.setState({
      position: { x, y },
      direction: face
    });
  };

  // move command
  movePacman = () => {
    const {
      direction,
      position: { x, y },
      gridNumber
    } = this.state;

    const newX = x + directionMove[direction].moveX;
    const newY = y + directionMove[direction].moveY;

    const newPosition = { x: newX, y: newY };
    if (!this.availablePosition(newPosition)) {
      // ErrorMsg: invalid move
      this.setState({
        message: `Invalid move, Please move position X and Y in 0 ~ ${gridNumber -
          1}`
      });
      return;
    }

    this.setState({
      position: newPosition
    });
  };

  // left command or right command
  rotatePacman = rotate => {
    const { direction } = this.state;
    const index = this.getIndexFromArray(direction, directions);

    let newDirection;
    if (rotate === "LEFT") {
      newDirection = directions[(index - 1 + 4) % 4];
    }
    if (rotate === "RIGHT") {
      newDirection = directions[(index + 1) % 4];
    }

    this.setState({
      direction: newDirection
    });
  };

  // report command
  report = () => {
    this.setState({
      isReport: true
    });
  };

  render() {
    const {
      gridNumber,
      direction,
      position,
      isReport,
      message,
      inputCommand
    } = this.state;

    return (
      <div>
        <h1 className="simulator__title">Pacman</h1>
        <div className="simulator__content">
          <div className="simulator__control">
            <InputArea
              handleChange={this.handleChange}
              onInputSubmit={this.onInputSubmit}
              inputCommand={inputCommand}
            />
            <button onClick={() => this.handleClick()}>RESET</button>
            {!!message ? (
              <p className="simulator__message">Message: {message}</p>
            ) : null}
            {isReport ? (
              <p className="simulator__output">
                Output: {position.x},{position.y},{direction}
              </p>
            ) : null}
          </div>
          <div className="simulator__map">
            <PacmanMap
              gridNumber={gridNumber}
              direction={direction}
              position={position}
            />
          </div>
        </div>
      </div>
    );
  }
}

export default Simulator;
