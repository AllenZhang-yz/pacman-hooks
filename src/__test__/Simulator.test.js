import React from "react";
import { mount } from "enzyme";
import Simulator from "../components/Simulator";
import InputArea from "../components/InputArea";
import PacmanMap from "../components/PacmanMap";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import toJson from "enzyme-to-json";
Enzyme.configure({ adapter: new Adapter() });

const setup = () => {
  const wrapper = mount(<Simulator />);
  return wrapper;
};

const directions = ["NORTH", "EAST", "SOUTH", "WEST"];

describe("simulator", () => {
  describe("Simulator renders correctly", () => {
    const wrapper = setup();
    it("Simulator renders correctly", () => {
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
  describe("Title", () => {
    const wrapper = setup();

    it("Simulator Title renders correctly", () => {
      const title = wrapper.find("h1");
      expect(title.length).toBe(1);
      expect(title.text()).toBe("Pacman");
    });
  });

  describe("InputArea", () => {
    let state = { inputCommand: "test" };
    const wrapper = setup();
    wrapper.setState(state);

    it("Simulator renders a InputArea", () => {
      expect(wrapper.find(InputArea).length).toBe(1);
    });
    it(`inputCommand is same to ${state.inputCommand}`, () => {
      expect(wrapper.find(InputArea).props().inputCommand).toBe(
        state.inputCommand
      );
    });
  });

  describe("RESET button", () => {
    const wrapper = setup();

    const mock_handleClick = jest.fn();
    const button = wrapper.find("button");
    const instance = wrapper.instance();
    it("Simulator renders a RESET button", () => {
      expect(button.text()).toBe("RESET");
    });

    it("button click", () => {
      instance.handleClick = mock_handleClick;
      button.simulate("click");
      expect(mock_handleClick).toBeCalled();
    });
  });

  describe("Message", () => {
    const wrapper = setup();

    const message = wrapper.find(".simulator__message");
    const instance = wrapper.instance();
    it("when have message", () => {
      const state = { message: "test message" };
      wrapper.setState(state);
      // expect(message.exists()).toEqual(true);
    });
  });

  describe("Output", () => {
    const wrapper = setup();

    const output = wrapper.find(".simulator__output");
    const instance = wrapper.instance();
    it("when have output", () => {
      const state = { isReport: true };
      wrapper.setState(state);
      // expect(message.exists()).toEqual(true);
    });
  });

  describe("PacmanMap", () => {
    const state = {
      gridNumber: 5,
      direction: "SOUTH",
      position: { x: 4, y: 0 }
    };
    const wrapper = setup();
    wrapper.setState(state);
    const pacmanMap = wrapper.find(PacmanMap);
    it("Simulator renders a PacmanMap", () => {
      expect(pacmanMap.length).toBe(1);
    });
    it("receive three props", () => {
      expect(Object.keys(pacmanMap.props()).length).toBe(3);
    });
    it("PacmanMap state passed", () => {
      expect(pacmanMap.props().gridNumber).toBe(5);
      expect(pacmanMap.props().direction).toBe("SOUTH");
      expect(pacmanMap.props().position.x).toBe(4);
      expect(pacmanMap.props().position.y).toBe(0);
    });
  });

  // check functions
  describe("check function", () => {
    const wrapper = setup();
    const instance = wrapper.instance();
    it("handleChange", () => {
      const spyFunction = jest.spyOn(instance, "handleChange");
      const value = "test";
      instance.handleChange(value);
      expect(spyFunction).toHaveBeenCalled();
      expect(wrapper.state("inputCommand")).toBe(value);
    });
    it("handleClick", () => {
      const spyFunction = jest.spyOn(instance, "handleClick");
      instance.handleClick();
      expect(spyFunction).toHaveBeenCalled();
    });

    describe("onInputSubmit", () => {
      const spyFunction = jest.spyOn(instance, "onInputSubmit");
      const event = { preventDefault: function() {} };
      const stateCase1 = { isReport: true };
      const stateCase2 = {
        isReport: false,
        position: { x: 7, y: 2 },
        inputCommand: "MOVE"
      };
      const stateCase3 = {
        isReport: false,
        position: { x: 7, y: 2 },
        inputCommand: "Place 2,3,North,sss"
      };
      const stateCase4 = {
        isReport: false,
        position: { x: 7, y: 2 },
        inputCommand: "Place 2,3,North"
      };
      const stateCase5 = {
        isReport: false,
        position: { x: 2, y: 2 },
        inputCommand: "MOVE"
      };
      const stateCase6 = {
        isReport: false,
        position: { x: 3, y: 0 },
        inputCommand: "LEFT"
      };
      const stateCase7 = {
        isReport: false,
        position: { x: 4, y: 0 },
        inputCommand: "Right"
      };
      const stateCase8 = {
        isReport: false,
        position: { x: 4, y: 0 },
        inputCommand: "REPORT"
      };
      const stateCase9 = {
        isReport: false,
        position: { x: 4, y: 0 },
        inputCommand: "testCommand"
      };

      it("onInputSubmit called stateCase1", () => {
        wrapper.setState(stateCase1);
        instance.onInputSubmit(event);
        expect(spyFunction).toHaveBeenCalled();
        expect(wrapper.state("message")).toBe(
          "Please click RESET button to start a new test!"
        );
      });
      it("onInputSubmit called stateCase2", () => {
        wrapper.setState(stateCase2);
        instance.onInputSubmit(event);
        expect(spyFunction).toHaveBeenCalled();
        expect(wrapper.state("message")).toBe(
          "Please enter place command FIRST!"
        );
      });
      it("onInputSubmit called stateCase3", () => {
        wrapper.setState(stateCase3);
        instance.onInputSubmit(event);
        expect(spyFunction).toHaveBeenCalled();
        expect(wrapper.state("message")).toBe("Invalid Command!");
      });
      it("onInputSubmit called stateCase4", () => {
        wrapper.setState(stateCase4);
        const spyPlacePacman = jest.spyOn(instance, "placePacman");
        instance.onInputSubmit(event);
        expect(spyFunction).toHaveBeenCalled();
        expect(spyPlacePacman).toHaveBeenCalled();
      });
      it("onInputSubmit called stateCase5", () => {
        wrapper.setState(stateCase5);
        const spyMovePacman = jest.spyOn(instance, "movePacman");
        instance.onInputSubmit(event);
        expect(spyFunction).toHaveBeenCalled();
        expect(spyMovePacman).toHaveBeenCalled();
      });
      it("onInputSubmit called stateCase6", () => {
        wrapper.setState(stateCase6);
        const spyRotatePacman = jest.spyOn(instance, "rotatePacman");
        instance.onInputSubmit(event);
        expect(spyFunction).toHaveBeenCalled();
        expect(spyRotatePacman).toHaveBeenCalled();
      });
      it("onInputSubmit called stateCase7", () => {
        wrapper.setState(stateCase7);
        const spyRotatePacman = jest.spyOn(instance, "rotatePacman");
        instance.onInputSubmit(event);
        expect(spyFunction).toHaveBeenCalled();
        expect(spyRotatePacman).toHaveBeenCalled();
      });
      it("onInputSubmit called stateCase8", () => {
        wrapper.setState(stateCase8);
        const spyReport = jest.spyOn(instance, "report");
        instance.onInputSubmit(event);
        expect(spyFunction).toHaveBeenCalled();
        expect(spyReport).toHaveBeenCalled();
      });
      it("onInputSubmit called stateCase9", () => {
        wrapper.setState(stateCase9);
        instance.onInputSubmit(event);
        expect(spyFunction).toHaveBeenCalled();
        expect(wrapper.state("message")).toBe("Invalid Command!");
      });
    });

    it("getIndexFromArray", () => {
      const spyFunction = jest.spyOn(instance, "getIndexFromArray");
      const arr = directions;
      const value = "EAST";
      instance.getIndexFromArray(value, arr);
      expect(spyFunction).toHaveBeenCalled();
      expect(spyFunction(value, arr)).toBe(1);
    });
    describe("availablePosition", () => {
      const spyFunction = jest.spyOn(instance, "availablePosition");
      const position1 = { x: -2, y: 2 };
      const position2 = { x: 7, y: 2 };
      const position3 = { x: 2, y: -2 };
      const position4 = { x: 4, y: 8 };
      const position5 = { x: 3, y: 2 };

      it("availablePosition position1", () => {
        instance.availablePosition(position1);
        expect(spyFunction).toHaveBeenCalled();
        expect(spyFunction(position1)).toBe(false);
      });
      it("availablePosition position2", () => {
        instance.availablePosition(position2);
        expect(spyFunction).toHaveBeenCalled();
        expect(spyFunction(position2)).toBe(false);
      });
      it("availablePosition position3", () => {
        instance.availablePosition(position3);
        expect(spyFunction).toHaveBeenCalled();
        expect(spyFunction(position3)).toBe(false);
      });
      it("availablePosition position4", () => {
        instance.availablePosition(position4);
        expect(spyFunction).toHaveBeenCalled();
        expect(spyFunction(position4)).toBe(false);
      });
      it("availablePosition position5", () => {
        instance.availablePosition(position5);
        expect(spyFunction).toHaveBeenCalled();
        expect(spyFunction(position5)).toBe(true);
      });
    });

    describe("placePacman", () => {
      const spyFunction = jest.spyOn(instance, "placePacman");
      const case1 = [-3, 3, "EAST"];
      const case2 = [3, 3, "test"];
      const case3 = [3, 3, "EAST"];
      it("placePacman case1", () => {
        instance.placePacman(case1);
        expect(spyFunction).toHaveBeenCalled();
        expect(wrapper.state("message")).toBe(
          "Invalid Position, Please place position X and Y in 0 ~ 4"
        );
      });
      it("placePacman case2", () => {
        instance.placePacman(case2);
        expect(spyFunction).toHaveBeenCalled();
        expect(wrapper.state("message")).toBe(
          "Invalid Direction, Please use NORTH, EAST, SOUTH, WEST"
        );
      });
      it("placePacman case3", () => {
        instance.placePacman(case3);
        expect(spyFunction).toHaveBeenCalled();
        expect(wrapper.state("position").x).toBe(3);
        expect(wrapper.state("position").y).toBe(3);
        expect(wrapper.state("direction")).toBe("EAST");
      });
    });
    describe("movePacman", () => {
      const spyFunction = jest.spyOn(instance, "movePacman");
      const stateCase1 = { direction: "WEST", position: { x: 0, y: 4 } };
      const stateCase2 = { direction: "WEST", position: { x: 1, y: 4 } };
      it("movePacman stateCase1", () => {
        wrapper.setState(stateCase1);
        instance.movePacman();
        expect(spyFunction).toHaveBeenCalled();
        expect(wrapper.state("message")).toBe(
          "Invalid move, Please move position X and Y in 0 ~ 4"
        );
      });
      it("movePacman stateCase2", () => {
        wrapper.setState(stateCase2);
        instance.movePacman();
        expect(spyFunction).toHaveBeenCalled();
        expect(wrapper.state("position").x).toBe(0);
        expect(wrapper.state("position").y).toBe(4);
      });
    });
    describe("rotatePacman", () => {
      const spyFunction = jest.spyOn(instance, "rotatePacman");
      const state = { direction: "NORTH" };
      it("rotatePacman LEFT", () => {
        wrapper.setState(state);
        instance.rotatePacman("LEFT");
        expect(spyFunction).toHaveBeenCalled();
        expect(wrapper.state("direction")).toBe("WEST");
      });
      it("rotatePacman RIGHT", () => {
        wrapper.setState(state);
        instance.rotatePacman("RIGHT");
        expect(spyFunction).toHaveBeenCalled();
        expect(wrapper.state("direction")).toBe("EAST");
      });
    });
    it("report", () => {
      const spyFunction = jest.spyOn(instance, "report");
      instance.report();
      expect(spyFunction).toHaveBeenCalled();
      expect(wrapper.state("isReport")).toBe(true);
    });
  });
});
