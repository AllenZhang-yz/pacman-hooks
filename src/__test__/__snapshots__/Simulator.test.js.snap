// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`simulator Simulator renders correctly Simulator renders correctly 1`] = `
<Simulator>
  <div>
    <h1
      className="simulator__title"
    >
      Pacman
    </h1>
    <div
      className="simulator__content"
    >
      <div
        className="simulator__control"
      >
        <InputArea
          handleChange={[Function]}
          inputCommand=""
          onInputSubmit={[Function]}
        >
          <form
            className="input-area"
            onSubmit={[Function]}
          >
            <input
              onChange={[Function]}
              placeholder="Please enter command"
              value=""
            />
          </form>
        </InputArea>
        <button
          onClick={[Function]}
        >
          RESET
        </button>
      </div>
      <div
        className="simulator__map"
      >
        <PacmanMap
          direction=""
          gridNumber={5}
          position={
            Object {
              "x": -1,
              "y": -1,
            }
          }
        >
          <div>
            <p>
              Grid Number: 
              5
            </p>
            <div
              className="pacman-map"
            >
              <div
                className="pacman-map__cell"
                key="0"
              >
                (0, 0)
              </div>
              <div
                className="pacman-map__cell"
                key="1"
              >
                (1, 0)
              </div>
              <div
                className="pacman-map__cell"
                key="2"
              >
                (2, 0)
              </div>
              <div
                className="pacman-map__cell"
                key="3"
              >
                (3, 0)
              </div>
              <div
                className="pacman-map__cell"
                key="4"
              >
                (4, 0)
              </div>
              <div
                className="pacman-map__cell"
                key="5"
              >
                (0, 1)
              </div>
              <div
                className="pacman-map__cell"
                key="6"
              >
                (1, 1)
              </div>
              <div
                className="pacman-map__cell"
                key="7"
              >
                (2, 1)
              </div>
              <div
                className="pacman-map__cell"
                key="8"
              >
                (3, 1)
              </div>
              <div
                className="pacman-map__cell"
                key="9"
              >
                (4, 1)
              </div>
              <div
                className="pacman-map__cell"
                key="10"
              >
                (0, 2)
              </div>
              <div
                className="pacman-map__cell"
                key="11"
              >
                (1, 2)
              </div>
              <div
                className="pacman-map__cell"
                key="12"
              >
                (2, 2)
              </div>
              <div
                className="pacman-map__cell"
                key="13"
              >
                (3, 2)
              </div>
              <div
                className="pacman-map__cell"
                key="14"
              >
                (4, 2)
              </div>
              <div
                className="pacman-map__cell"
                key="15"
              >
                (0, 3)
              </div>
              <div
                className="pacman-map__cell"
                key="16"
              >
                (1, 3)
              </div>
              <div
                className="pacman-map__cell"
                key="17"
              >
                (2, 3)
              </div>
              <div
                className="pacman-map__cell"
                key="18"
              >
                (3, 3)
              </div>
              <div
                className="pacman-map__cell"
                key="19"
              >
                (4, 3)
              </div>
              <div
                className="pacman-map__cell"
                key="20"
              >
                (0, 4)
              </div>
              <div
                className="pacman-map__cell"
                key="21"
              >
                (1, 4)
              </div>
              <div
                className="pacman-map__cell"
                key="22"
              >
                (2, 4)
              </div>
              <div
                className="pacman-map__cell"
                key="23"
              >
                (3, 4)
              </div>
              <div
                className="pacman-map__cell"
                key="24"
              >
                (4, 4)
              </div>
            </div>
          </div>
        </PacmanMap>
      </div>
    </div>
  </div>
</Simulator>
`;

exports[`simulator Simulator renders correctly Simulator renders correctly 2`] = `
<Simulator>
  <div>
    <h1
      className="simulator__title"
    >
      Pacman
    </h1>
    <div
      className="simulator__content"
    >
      <div
        className="simulator__control"
      >
        <InputArea
          handleChange={[Function]}
          inputCommand=""
          onInputSubmit={[Function]}
        >
          <form
            className="input-area"
            onSubmit={[Function]}
          >
            <input
              onChange={[Function]}
              placeholder="Please enter command"
              value=""
            />
          </form>
        </InputArea>
        <button
          onClick={[Function]}
        >
          RESET
        </button>
      </div>
      <div
        className="simulator__map"
      >
        <PacmanMap
          direction=""
          gridNumber={5}
          position={
            Object {
              "x": -1,
              "y": -1,
            }
          }
        >
          <div>
            <p>
              Grid Number: 
              5
            </p>
            <div
              className="pacman-map"
            >
              <div
                className="pacman-map__cell"
                key="0"
              >
                (0, 0)
              </div>
              <div
                className="pacman-map__cell"
                key="1"
              >
                (1, 0)
              </div>
              <div
                className="pacman-map__cell"
                key="2"
              >
                (2, 0)
              </div>
              <div
                className="pacman-map__cell"
                key="3"
              >
                (3, 0)
              </div>
              <div
                className="pacman-map__cell"
                key="4"
              >
                (4, 0)
              </div>
              <div
                className="pacman-map__cell"
                key="5"
              >
                (0, 1)
              </div>
              <div
                className="pacman-map__cell"
                key="6"
              >
                (1, 1)
              </div>
              <div
                className="pacman-map__cell"
                key="7"
              >
                (2, 1)
              </div>
              <div
                className="pacman-map__cell"
                key="8"
              >
                (3, 1)
              </div>
              <div
                className="pacman-map__cell"
                key="9"
              >
                (4, 1)
              </div>
              <div
                className="pacman-map__cell"
                key="10"
              >
                (0, 2)
              </div>
              <div
                className="pacman-map__cell"
                key="11"
              >
                (1, 2)
              </div>
              <div
                className="pacman-map__cell"
                key="12"
              >
                (2, 2)
              </div>
              <div
                className="pacman-map__cell"
                key="13"
              >
                (3, 2)
              </div>
              <div
                className="pacman-map__cell"
                key="14"
              >
                (4, 2)
              </div>
              <div
                className="pacman-map__cell"
                key="15"
              >
                (0, 3)
              </div>
              <div
                className="pacman-map__cell"
                key="16"
              >
                (1, 3)
              </div>
              <div
                className="pacman-map__cell"
                key="17"
              >
                (2, 3)
              </div>
              <div
                className="pacman-map__cell"
                key="18"
              >
                (3, 3)
              </div>
              <div
                className="pacman-map__cell"
                key="19"
              >
                (4, 3)
              </div>
              <div
                className="pacman-map__cell"
                key="20"
              >
                (0, 4)
              </div>
              <div
                className="pacman-map__cell"
                key="21"
              >
                (1, 4)
              </div>
              <div
                className="pacman-map__cell"
                key="22"
              >
                (2, 4)
              </div>
              <div
                className="pacman-map__cell"
                key="23"
              >
                (3, 4)
              </div>
              <div
                className="pacman-map__cell"
                key="24"
              >
                (4, 4)
              </div>
            </div>
          </div>
        </PacmanMap>
      </div>
    </div>
  </div>
</Simulator>
`;
